### PR TITLE
fix: [3] fix the error handling of getCommitSha

### PR DESCRIPTION
**Pull request by itleigns**
## Context

If we run a job in a pipeline whose corresponding SCM branch has been deleted, 500 error is thrown.
It is appropriate to throw 404 error in this case so we want to change that way.
## Objective

- We change the error status the `/event` endpoint throws when we run a job in a pipeline whose corresponding SCM branch has been deleted. We set error status as 404.
- We add a test for it.
- We eliminated unnecessary variable assignments in test code for refactoring.

## Context
~~We change the format of the error returned by `getCommitSha` in this [PR](https://github.com/screwdriver-cd/scm-github/pull/186) so we change the error handling of `getCommitSha` accordingly~~.

## Objective
- ~~We change the error handling of `getCommitSha`.~~
- ~~We change a test to follow this [PR](https://github.com/screwdriver-cd/scm-github/pull/186).~~
- ~~We add a test in case we run a pipeline with the corresponding SCM branch deleted.~~
- ~~We eliminated unnecessary variable assignments in test code for refactoring.~~

## Reference
https://github.com/screwdriver-cd/scm-gitlab/pull/36/
https://github.com/screwdriver-cd/scm-bitbucket/pull/75

## License
I confirm that this contribution is made under a BSD license and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: test/plugins/events.test.js [hunk 1]
```diff
@@ -748,6 +741,18 @@ describe('event plugin test', () => {
             });
         });
 
+        it('returns 404 when the model encounters an branch not found error', () => {
```
Comment (tkyi) [inline test/plugins/events.test.js:744]:
```suggestion
        it('returns 404 when the model encounters a branch not found error', () => {
```

**Review by tkyi — APPROVED** [review]
Small suggestion, otherwise LGTM

**Review by MysticDoll — CHANGES_REQUESTED** [review]
Did you read another scm component?
For example,  `getCommitSha` in scm-gitlab checks response by [checkResponseError](https://github.com/screwdriver-cd/scm-gitlab/blob/e2330660aaea4a8c9aa09e3e6d56a7a9d3db7fbd/index.js#L50-L66). And this function set `response.statusCode` to  `error.code` and also set `response.body.message` to `error.message`. Therefore the error object from scm-gitlab does **not** have `status` member and it is **not** guaranteed that `error.message` always has `Branch not found` in this case.

I think you should make the interfaces of errors in scm component common by using `boom.notFound`.

File: plugins/events/create.js [hunk 2]
```diff
@@ -219,9 +219,16 @@ module.exports = () => ({
...
-                                                payload.sha = sha;
-                                            })
+                                            scm
+                                                .getCommitSha(scmConfig)
+                                                .then(sha => {
+                                                    payload.sha = sha;
+                                                })
+                                                .catch(err => {
+                                                    if (err.status === 404) {
+                                                        throw boom.notFound(err.message);
```
Comment (MysticDoll) [inline plugins/events/create.js:229]:
```suggestion
                                                        throw boom.notFound(err);
```

to keep the information of the original error.

File: plugins/events/create.js [hunk 3]
```diff
@@ -219,9 +219,16 @@ module.exports = () => ({
...
-                                                payload.sha = sha;
-                                            })
+                                            scm
+                                                .getCommitSha(scmConfig)
+                                                .then(sha => {
+                                                    payload.sha = sha;
+                                                })
+                                                .catch(err => {
+                                                    if (err.status === 404) {
+                                                        throw boom.notFound(err);
```
Comment (sugarnaoming) [inline plugins/events/create.js:229]: If you use the following code, you may be able to handle errors other than 404 with boom.
`throw boom.boomify(err, {statusCode: err.status})`

Comment (kkisic) [reply plugins/events/create.js]: Thx. I fixed it.

**Review by kkokufud — APPROVED** [review]
LGTM

**Review by tkyi — APPROVED** [review]

**Pull request merged**